bootstrap/handlers/secret/secret.go

```diff
@@ -52,16 +53,38 @@ func (s *SecretProvider) BootstrapHandler(
 
 	lc := container.LoggingClientFrom(dic.Get)
 
-	configuration := container.ConfigurationFrom(dic.Get)
-	secretConfig, err := s.getSecretConfig(configuration.GetBootstrap().SecretStore)
-	if err != nil {
-		lc.Error(fmt.Sprintf("unable to parse secret store configuration: %s", err.Error()))
-		return false
-	}
-
 	// attempt to create a new SecretProvider client only if security is enabled.
 	if s.isSecurityEnabled() {
-		s.secretClient, err = client.NewVault(ctx, secretConfig, lc).Get(configuration.GetBootstrap().SecretStore)
+		var err error
+
+		lc.Info("Creating SecretClient")
+
+		configuration := container.ConfigurationFrom(dic.Get)
+		secretStoreConfig := configuration.GetBootstrap().SecretStore
+		retryWaitPeriod, err := time.ParseDuration(secretStoreConfig.RetryWaitPeriod)
+		if err != nil {
+			lc.Error(fmt.Sprintf("unable to parse RetryWaitPeriod: %s", err.Error()))
+			return false
+		}
+
+		for i := 1; i <= secretStoreConfig.AdditionalRetryAttempts+1; i++ {
+			lc.Info(fmt.Sprintf("Attempting to create secret client (attempt %d of %d)", i, secretStoreConfig.AdditionalRetryAttempts+1))
+
+			secretConfig, err := s.getSecretConfig(secretStoreConfig)
+			if err != nil {
+				lc.Error(fmt.Sprintf("unable to parse secret store configuration: %s", err.Error()))
+			} else {
+				secretClient, err := client.NewVault(ctx, secretConfig, lc).Get(configuration.GetBootstrap().SecretStore)
+				if err == nil {
+					s.secretClient = secretClient
+					lc.Info("Created SecretClient")
+					break
+				}
+			}
+
+			lc.Error(fmt.Sprintf("Retryable failure (after %s delay) while creating SecretClient: %s", secretStoreConfig.RetryWaitPeriod, err.Error()))
```

Comment:
`err` at this point may be `nil`. Should we add a `nil` check or move this into the previous `if-else` statement?

Reply:
if err is nil at line 74, we execute the else branch.
if err is nil at line 78, we break out of the loop and never get to line 85

What is the scenario where error is nil when we get here?

Reply:
Disregard. Whole new implementation.